Frameworks/Foundation/NSUndoManager.mm

```diff
@@ -16,16 +16,69 @@
 
 #import <StubReturn.h>
 #import <Foundation/NSUndoManager.h>
+#import <Foundation/NSMutableArray.h>
+#import "_NSUndoCall.h"
+#import "_NSUndoGroup.h"
 
-NSString* const NSUndoManagerGroupIsDiscardableKey = @"NSUndoManagerGroupIsDiscardableKey";
+@implementation NSUndoManager {
+    NSMutableArray<_NSUndoGroup*>* _undoStack;
+    NSMutableArray<_NSUndoGroup*>* _redoStack;
+}
 
-@implementation NSUndoManager
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
+*/
+- (id)init {
+    if (self = [super init]) {
+        _groupsByEvent = YES;
+        _NSUndoGroup* startingGroup = [[[_NSUndoGroup alloc] initWithOwner:self] autorelease];
+        _NSUndoGroup* startingRedoGroup = [[[_NSUndoGroup alloc] initWithOwner:self] autorelease];
+
+        _undoStack = [NSMutableArray array];
+        _redoStack = [NSMutableArray array];
+
+        [_undoStack addObject:startingGroup];
+        [_redoStack addObject:startingRedoGroup];
+
+        _undoRegistrationEnabled = YES;
+    }
+    return self;
+}
+
+/**
+ @Status Interoperable
 */
 - (void)registerUndoWithTarget:(id)target selector:(SEL)aSelector object:(id)anObject {
-    UNIMPLEMENTED();
+    if (!(self.isUndoRegistrationEnabled)) {
+        return;
+    }
+
+    _NSUndoCall* undoCall = [[[_NSUndoCall alloc] _initWithTarget:target selector:aSelector object:anObject] autorelease];
+    [self _registerUndoAction:undoCall];
```

Comment:
nit: since this is so short, it'd probably be better to just flip the if and put this inside rather than return